src/vs/editor/contrib/dnd/browser/dndHintWidget.ts

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as editorBrowser from 'vs/editor/browser/editorBrowser';
+import { FastDomNode, createFastDomNode } from 'vs/base/browser/styleMutator';
+import { Widget } from 'vs/base/browser/ui/widget';
+import { IDisposable, dispose } from 'vs/base/common/lifecycle';
+import { Configuration } from 'vs/editor/browser/config/configuration';
+import { Position } from 'vs/editor/common/core/position';
+import { IPosition, TextEditorCursorStyle, IConfigurationChangedEvent } from 'vs/editor/common/editorCommon';
+
+export class DragTargetHintWidget extends Widget implements editorBrowser.IContentWidget {
+
+	static ID = 'editor.contrib.dragTargetHintWidget';
+	protected _editor: editorBrowser.ICodeEditor;
+	protected _showAtPosition: Position;
+	private disposables: IDisposable[] = [];
+	private _cursorStyle: TextEditorCursorStyle;
+	private _lineHeight: number;
+	private _typicalHalfwidthCharacterWidth: number;
+	private readonly _domNode: FastDomNode;
+	private _isVisible: boolean = false;
+
+	protected get isVisible(): boolean {
+		return this._isVisible;
+	}
+
+	protected set isVisible(value: boolean) {
+		this._isVisible = value;
+	}
+
+	constructor(editor: editorBrowser.ICodeEditor) {
+		super();
+		this._editor = editor;
+
+
+		// Create the dom node
+		this._domNode = createFastDomNode(document.createElement('div'));
+		this._domNode.setClassName('cursor secondary');
+		this._domNode.setTop(0);
+		this._domNode.setLeft(0);
+		this._domNode.setAttribute('role', 'presentation');
+		this._domNode.setAttribute('aria-hidden', 'true');
+		this._domNode.setVisibility('hidden');
+		this._editor.addContentWidget(this);
+
+		this._cursorStyle = this._editor.getConfiguration().viewInfo.cursorStyle;
```

Comment:
I don't like that we reuse here the cursor style and the cursor CSS. It makes it difficult to make changes in the CSS and e.g. it makes no sense to use a block cursor to paint the insertion point.

IMHO the insertion point should always be rendered as a line, regardless of user settings. It is not a cursor, it is an insertion point.

On top, I would render the insertion point in a distinct way than the cursor. I got very confused when I tried it out, it looks to much like a cursor, it should be orange or something to let me know that it represents the drop point and is not some editor bug where a cursor is painted.

It should also go away if the mouse is not sitting directly on text (mouseEvent.target !== CONTENT). If the drag is on top of a view zone, the margin, etc. there should be no drop. IMHO, only on top of content.